elkodon/tests/publisher_tests.rs

```diff
@@ -10,128 +10,162 @@ mod publisher {
     use elkodon_bb_posix::unique_system_id::UniqueSystemId;
     use elkodon_bb_testing::assert_that;
 
+    type TestResult<T> = core::result::Result<T, Box<dyn std::error::Error>>;
+
     const TIMEOUT: Duration = Duration::from_millis(25);
 
-    fn generate_name() -> ServiceName {
-        let mut service = ServiceName::new(b"service_tests_").unwrap();
-        service
-            .push_bytes(
-                UniqueSystemId::new()
-                    .unwrap()
-                    .value()
-                    .to_string()
-                    .as_bytes(),
-            )
-            .unwrap();
-        service
+    fn generate_name() -> TestResult<ServiceName> {
+        let mut service = ServiceName::new(b"service_tests_")?;
+        service.push_bytes(
+            UniqueSystemId::new()
+                .unwrap()
```

Comment:
Unfortunately I could not replace all `unwrap` in this file since some types do not yet implement the `Error` trait.